### PR TITLE
feat: add AVS guide

### PR DESCRIPTION
**Pull request by aroralanuk**
fixes: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3601

### CODE REVIEW

File: docs/guides/avs-operator-guide.mdx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+cast send --private-key <PRIVATE_KEY> $SERVICE_MANAGER "enrollIntoChallenger(address)" <CHALLENGER_ADDRESS> --rpc-url <RPC_URL>
+```
+
+NOTE: The IRemoteChallenger implementations are not yet live in production.
+
+## Register your operator
+
+:::warning
+
+There is a distinction between the operator key and the AVS signing key. The operator key is meant for just interacting with the AVS contracts whereas the AVS signing key is used for active operations which in our case will be signing checkpoints with a Hypelane validator. As on May 15th, 2024. Eigenlayer doesn't support attesting to an AVS signing key while registering as an operator but that will be changing in short order. More details about the keys [here](https://docs.eigenlayer.xyz/eigenlayer/operator-guides/key-management/intro#operator-keys).
```
Comment (paulbalaji) [inline docs/guides/avs-operator-guide.mdx:54]:
```suggestion
There is a distinction between the operator key and the AVS signing key. The operator key is meant for just interacting with the AVS contracts whereas the AVS signing key is used for active operations, which in our case will be signing checkpoints with a Hyperlane Validator. As of May 15th 2024, EigenLayer doesn't support attesting to an AVS signing key while registering as an operator but that will be changing in short order. You can read more details about the keys [here](https://docs.eigenlayer.xyz/eigenlayer/operator-guides/key-management/intro#operator-keys).
```

File: docs/guides/avs-operator-guide.mdx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+:::
+
+For registering your operator to Hyperlane AVS, you can register your operator key generated using Eignelayer CLI by creating an EIP-191 compliant signature and submitting it via our CLI:
```
Comment (paulbalaji) [inline docs/guides/avs-operator-guide.mdx:58]:
```suggestion
To register your operator to Hyperlane AVS, you can register your operator key (generated via EigenLayer CLI) by creating an EIP-191 compliant signature and submitting it via the Hyperlane CLI:
```

File: docs/guides/avs-operator-guide.mdx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+```bash
+ hyperlane avs register --operatorKeyPath <PATH_TO_OPERATOR_KEY> --chain holesky --key <PRIVATE_KEY_FOR_SENDING_TX>
+```
+
+Note: PRIVATE_KEY_FOR_SENDING_TX can be any third-party key and doesn't need to be operator key.
```
Comment (paulbalaji) [inline docs/guides/avs-operator-guide.mdx:64]: admonition again

File: docs/protocol/avs.mdx [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Hyperlane AVS
+
+### Protocol SLA
```
Comment (nambrot) [inline docs/protocol/avs.mdx:3]: I would either be a lot more elaborate or remote this page. Right now, as somebody who doesn't know anything about the Hyperlane AVS, I have no idea about the purpose of the AVS, what a challenger is, etc.

Comment (aroralanuk) [reply docs/protocol/avs.mdx:3]: Ok, I'll remove this and add this section to a new PR where I'll add more details. Don't want to block operators looking at the guide.

File: docs/guides/avs-operator-guide.mdx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+You can monitor Eigenlayer specific endpoints via the `--metrics` port you specified in the validator configuration. Read more about the example endpoints and what should be expected [here](https://docs.eigenlayer.xyz/eigenlayer/avs-guides/spec/api/).
+
+The supported endpoints are:
+
+| Endpoint                                  | Description                                |
+| ----------------------------------------- | ------------------------------------------ |
+| **GET /eigen/node**                       | for node info                              |
+| **GET /eigen/node/health**                | for node health                            |
+| **GET /eigen/node/services**              | for a list of services the node is running |
+| **GET /eigen/node/`{service_id}`/health** | for health of a specific service           |
```
Comment (nambrot) [inline docs/guides/avs-operator-guide.mdx:38]: What is `service_id` here? IIRC, we didn't really have subservices, so maybe lets just omit this information?

**Review by cmcewen — APPROVED** [review]
LGTM once @nambrot's comments are addressed

File: docs/guides/avs-operator-guide.mdx (hunk 5 above)
Comment (aroralanuk) [reply docs/guides/avs-operator-guide.mdx:38]: We have serviceId `hyperlane-validator-indexer` and `hyperlane-validator-submitter` hardcoded. So, I'll remove it until it's not hardcoded.

**Pull request merged**